### PR TITLE
Adding OAuth2 Provider for Keap

### PR DESCRIPTION
**Pull request by brumiser1550**
# Submitting Pull Requests

## General

 - [x] Make sure you use [semantic commit messages](https://seesparkbox.com/foundry/semantic_commit_messages).
       Examples: `"fix(google): Fixed foobar bug"`, `"feat(accounts): Added foobar feature"`.
 - [x] All Python code must formatted using Black, and clean from pep8 and isort issues.
 - [x] JavaScript code should adhere to [StandardJS](https://standardjs.com).
 - [x] If your changes are significant, please update `ChangeLog.rst`.
 - [x] If your change is substantial, feel free to add yourself to `AUTHORS`.

 ## Provider Specifics

 In case you add a new provider:

- [x] Make sure unit tests are available.
- [x] Add an entry of your provider in `test_settings.py::INSTALLED_APPS` and `docs/installation.rst::INSTALLED_APPS`.
- [x] Add documentation to `docs/providers.rst`.
- [x] Add an entry to the list of supported providers over at `docs/overview.rst`.

### CODE REVIEW

File: allauth/socialaccount/providers/keap/views.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    access_token_url = "https://api.infusionsoft.com/token"
+    authorize_url = "https://accounts.infusionsoft.com/app/oauth/authorize"
+    profile_url = "https://api.infusionsoft.com/crm/rest/v1/oauth/connect/userinfo"
+
+    def complete_login(self, request, app, token, **kwargs):
+        response = kwargs.get("response", None)
+        scope = response.get("scope")
+        app_name = scope.split("|")[1].split(".")[0]
+
+        extra_data = requests.get(
```
Comment (pennersr) [inline allauth/socialaccount/providers/keap/views.py:26]: Can you split this up in something like?:

```
response = requests.get(...)
response.raise_for_status()
extra_data = response.json()
```

Comment (brumiser1550) [reply allauth/socialaccount/providers/keap/views.py]: @pennersr Done, let me know if I should address anything else. Thanks for taking a look!

Comment (pennersr) [reply allauth/socialaccount/providers/keap/views.py]: If you can have a look at the test suite -- it is currently failing.

Comment (brumiser1550) [reply allauth/socialaccount/providers/keap/views.py]: Ok, I think I have gotten what was needed. Sorry for the back and forth, pretty new to testing/contributing.

File: allauth/socialaccount/providers/keap/views.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        response = kwargs.get("response", {})
+        scope = response.get("scope")
+        app_name = scope.split("|")[1].split(".")[0]
+        profile_response = requests.get(
+            self.profile_url, params={"access_token": token.token}
+        )
+        profile_response.raise_for_status()
+        extra_data = profile_response.json()
+        extra_data["scope"] = scope
+        extra_data["app_name"] = app_name
```
Comment (pennersr) [inline allauth/socialaccount/providers/keap/views.py:30]: **Issue**: The `complete_login()` method is not supposed to return `None` --  it is supposed to return ` SocialLogin` instance.

**Pull request closed without merging**